PedroChaparro/loomies-backend
# Feat: near loomies

## Includes: 

- [x] Update user struct to add the last time a loomie was generated by the user and the current generation timeout. 
- [x] Check if the user can generate new loomies according to the mentioned above fields.
- [x] Generate loomies near the user area.
- [x] Update the "last time user generated loomies" and "current generation timeout" fields.
- [x] Script to remove the outdated loomies. 

## Notes

**Please,** check the new .env.config files and update your .env files before testing this PR. 

These are the "suggested" values for the .env file: 

```bash
GAME_MIN_LOOMIES_GENERATION_TIMEOUT = 2
GAME_MAX_LOOMIES_GENERATION_TIMEOUT = 10
GAME_MIN_LOOMIES_GENERATION_AMOUNT = 6
GAME_MAX_LOOMIES_GENERATION_AMOUNT = 16
GAME_LOOMIES_GENERATION_RADIUS = 0.00175
GAME_MAX_LOOMIES_PER_ZONE = 16
```

## How can I test it?

1. POST the `/near_loomies` endpoint with the latitude and longitude as JSON payload. You should get an array with wild loomies if the coordinates are "supported". 
2. POST the endpoint again and you should get the same loomies array (Until the timeout ends). 
3. POST the endpoint with the same coordinates but  logged-in as another user, you should get the previous loomies plus the new generated ones. 
4. Run the `clear:all` command from `algorithms/database_setup`. All the wild loomies should be deleted from the database, you can check it by using Mongo Express or Mongo Compass.
5. Wait for the timeout (Plus some seconds) and POST the endpoint again, then, wait "OUTDATED_LOOMIES_TIMEOUT" minutes and run the `clear:outdated` command from `algorithms/database_setup`.
